ocaml/xapi/repository_helpers.ml

```diff
@@ -733,7 +689,11 @@ let get_pkgs_from_repoquery pkg_narrow repositories =
 let get_updates_from_repoquery repositories =
   List.iter (fun r -> clean_yum_cache r) repositories ;
   (* Use 'updates' to decrease the number of packages to apply 'is_obsoleted' *)
-  let updates = get_pkgs_from_repoquery "updates" repositories in
+  let open Pkg_mgr in
+  let updates_arg =
+    match active () with Yum -> "updates" | Dnf -> "upgrades"
+  in
```

Comment:
Please move this into `get_pkgs_from_repoquery`

Reply:
No, it can not.
The `get_pkgs_from_repoquery` following this line is defined in `repository_helpers.ml`, 
and this is just a sub-command provided by Pkg_mgr.get_pkgs_from_repoquery
There are callings like `get_pkgs_from_repoquery "available"` to query available packages. 
Pkg_mgr does not know what the sub-command is. (unless Pkg_mgr provides an interface for every sub-command, but I doubt we need this)

Reply:
Well, after re-consider this, I decided to follow the suggestion to add interface for sub-command.